### PR TITLE
tsdb: improve logs when encountering corruption

### PR DESCRIPTION
**Pull request by simonpasquier**
I've spotted a few places where logs and errors could be improved while intentionally breaking my local tsdb...

cc @codesome, @krasi-georgiev

### CODE REVIEW

File: tsdb/db.go [hunk 1]
```diff
@@ -880,7 +884,11 @@ func (db *DB) reload() (err error) {
 				block.Close()
 			}
 		}
-		return fmt.Errorf("unexpected corrupted block:%v", corrupted)
+		var merr tsdb_errors.MultiError
+		for ulid, err := range corrupted {
+			merr.Add(errors.Wrap(err, ulid.String()))
+		}
+		return merr.Err()
```
Comment (bwplotka) [inline tsdb/db.go:891]: I think we lost wrapping here.. should we had errors.Wrap here as well?

File: tsdb/db.go [hunk 2]
```diff
@@ -441,10 +441,14 @@ func (db *DBReadOnly) Blocks() ([]BlockReader, error) {
...
-				level.Warn(db.logger).Log("msg", "Closing a block", err)
+				level.Warn(db.logger).Log("msg", "Closing block failed", "err", err, "block", b)
 			}
 		}
-		return nil, errors.Errorf("unexpected corrupted block:%v", corrupted)
+		var merr tsdb_errors.MultiError
+		for ulid, err := range corrupted {
+			merr.Add(errors.Wrap(err, ulid.String()))
+		}
+		return nil, merr.Err()
```
Comment (bwplotka) [inline tsdb/db.go:451]: ditto

Comment (simonpasquier) [reply tsdb/db.go:451]: done

File: tsdb/db.go (hunk 1 above)
Comment (simonpasquier) [reply tsdb/db.go:891]: done

**Pull request merged**